cirq/ops/common_channels.py

```diff
@@ -187,3 +189,528 @@ def depolarize(p: float) -> DepolarizingChannel:
+            return NotImplemented
+        return self._eq_tuple() == other._eq_tuple()
+
+    def __ne__(self, other):
+        return not self == other
+
+    def __repr__(self) -> str:
+        return 'cirq.phase_damping(gamma={!r})'.format(self._gamma)
+
+    def __str__(self) -> str:
+        return 'phase_damping(gamma={!r})'.format(self._gamma)
+
+    def _circuit_diagram_info_(
+        self, args: protocols.CircuitDiagramInfoArgs
+    ) -> str:
+        return 'PD({!r})'.format(self._gamma)
+
+
+def phase_damping(gamma: float) -> PhaseDampingChannel:
+    """
+    Creates a phase damping channel with damping rate gamma
+
+    This channel evovles a density matrix via
+           \rho -> M_0 \rho M_0^\dagger + M_1 \rho M_1^\dagger
+
+    With
+        M_0 = \begin{bmatrix} 1 & 0  \\ 0 & \sqrt{1 - \gamma} \end{bmatrix}
+        M_1 = \begin{bmatrix} 0 & 0 \\ 0 & \sqrt{\gamma} \end{bmatrix}
+
+    Args:
+        gamma: The damping rate
+
+    Raises:
+        ValueError is gamma is not a valid probability.
+    """
+    return PhaseDampingChannel(gamma)
+
+
+class PhaseFlipChannel(raw_types.Gate):
+    """Channel to flip a qubit's Phase"""
+
+    def __init__(self, p) -> None:
+
+        """
+        This channel flips a qubit's Phase with probability p
+
+        This channel evovles a density matrix via
+            \rho -> M_0 \rho M_0^\dagger + M_1 \rho M_1^\dagger
+
+        With
+            M_0 = \sqrt{p} \begin{bmatrix} 1 & 0  \\ 0 & 1 \end{bmatrix}
+            M_1 = \sqrt{1-p} \begin{bmatrix} 1 & 0 \\ 0 & -1 \end{bmatrix}
+
+        Args:
+            p: the probability of a phase flip
+
+        Raises:
+            ValueError if p is not a valid probability.
+        """
+
+        def validate_probability(p, p_str):
+            if p < 0:
+                raise ValueError('{} was less than 0.'.format(p_str))
+            elif p > 1:
+                raise ValueError('{} was greater than 1.'.format(p_str))
+            return p
+
+        self._p = validate_probability(p, 'p')
+        self._delegate = AsymmetricDepolarizingChannel(0.0, 0.0, 1.0 - p)
+
+    def _channel_(self) -> Iterable[np.ndarray]:
+        kraus_ops = list(self._delegate._channel_())
+        # just return identity and z term
+        return (kraus_ops[0], kraus_ops[3])
+
+    def _eq_tuple(self):
+        return (PhaseFlipChannel, self._p)
+
+    def __eq__(self, other):
+        if not isinstance(other, type(self)):
+            return NotImplemented
+        return self._eq_tuple() == other._eq_tuple()
+
+    def __ne__(self, other):
+        return not self == other
+
+    def __repr__(self) -> str:
+        return 'cirq.phase_flip(p={!r})'.format(self._p)
+
+    def __str__(self) -> str:
+        return 'phase_flip(p={!r})'.format(self._p)
+
+    def _circuit_diagram_info_(
+        self, args: protocols.CircuitDiagramInfoArgs
+    ) -> str:
+        return 'PF({!r})'.format(self._p)
+
+
+def phase_flip(p: float) -> PhaseFlipChannel:
```

Comment:
I would have assumed that this was the unitary Z gate, and been very surprised about there being noisy-ness.

Perhaps what we could do is have `p` default to 1, actually return a Z gate if p equals 1, and rename `p` to `probability_of_apply` or something descriptive like that. The method's return type would change to `Union[ops.ZPowGate, PhaseFlipChannel]` and you would also declare an overload that returned `ops.ZPowGate` when there is no  `probability_of_apply` argument.

A probably-even-better alternative would be to encapsulate this entire paradigm into a "ProbableOperation" that you could wrap around any operation (whether unitary or channel). We could then add a `with_probability(p)` method to `cirq.Operation` and `cirq.Gate` to make these easy to make, so you could say `cirq.Z(q).with_probability(p)`.

Reply:
I agree we need a better structure here but am not sure that this is the correct pull request to do that?  Righting these in the first place is an exercise to see what the structure is that we need.  The name is not horrible as a first take, but see where there could be confusion.

Reply:
I feel like this is something that should be tackled in separate PR. This is because some of the models in here have "probabilities", but others also have "rates", so the correct design of `with_probability` might not be so straightforward.

Reply:
Okay. Open an issue (you don't have to take it, just make it) to make a generalized unitary-to-probabilistic-unitary method and that is sufficient for now.